pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/AdminResource.java

```diff
@@ -624,16 +621,18 @@ protected boolean isNamespaceReplicated(NamespaceName namespaceName) {
         try {
             String partitionedTopicPath = path(PARTITIONED_TOPIC_PATH_ZNODE,
                     namespaceName.toString(), topicDomain.value());
-            List<String> topics = namespaceResources().getChildren(partitionedTopicPath);
+            List<String> topics = namespaceResources().getStore().getChildren(partitionedTopicPath).get();
             partitionedTopics = topics.stream()
                     .map(s -> String.format("%s://%s/%s", topicDomain.value(), namespaceName.toString(), decode(s)))
                     .collect(Collectors.toList());
-        } catch (NotFoundException e) {
-            // NoNode means there are no partitioned topics in this domain for this namespace
         } catch (Exception e) {
-            log.error("[{}] Failed to get partitioned topic list for namespace {}", clientAppId(),
-                    namespaceName.toString(), e);
-            throw new RestException(e);
+            if (e instanceof ExecutionException && e.getCause() instanceof NotFoundException) {
+                // NoNode means there are no partitioned topics in this domain for this namespace
+            } else {
+                log.error("[{}] Failed to get partitioned topic list for namespace {}", clientAppId(),
+                        namespaceName.toString(), e);
+                throw new RestException(e);
```

Comment:
Would it be better, just a suggestion
`throw new RestException(e instanceof ExecutionException ? e.getCause() : e)`